tracer/src/Datadog.Trace/AppSec/Waf/Result.cs

```diff
@@ -24,21 +24,19 @@ public Result(DdwafResultStruct returnStruct, DDWAF_RET_CODE returnCode, WafNati
             this.wafNative = wafNative;
             Actions = new string[returnStruct.ActionsSize];
             ReadActions(returnStruct);
-
+            Block = Actions.Contains("block");
+            ShouldBeReported = returnCode >= DDWAF_RET_CODE.DDWAF_MATCH;
             AggregatedTotalRuntime = aggregatedTotalRuntime;
             AggregatedTotalRuntimeWithBindings = aggregatedTotalRuntimeWithBindings;
         }
 
-        ~Result()
-        {
-            Dispose(false);
-        }
+        ~Result() => Dispose(false);
```

Comment:
I know this isn't a change made, but do you definitely need a finalizer? It puts added pressure on the GC. Can we not dispose the `Result` ourselves?

Reply:
good point!!

Reply:
removed in 60213612bd9dc48fe4f254b95d896f2a95e759d4